packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,864 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Do not import this file in production applications or packages published
+// to pub.dev. Flutter will make breaking changes to this file, even in patch
+// versions.
+//
+// All APIs in this file must be private or must:
+//
+// 1. Have the `@internal` attribute.
+// 2. Throw an  `UnsupportedError` if `isWindowingEnabled`
+//    is `false.
+//
+// See: https://github.com/flutter/flutter/issues/30701.
+
+import 'dart:ui' show AppExitType, Display, FlutterView;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import '../foundation/_features.dart';
+import 'binding.dart';
+import 'framework.dart';
+import 'view.dart';
+
+const String _windowingDisabledErrorMessage = '''
+Windowing APIs are not enabled.
+Windowing APIs are currently experimental. Do not use windowing APIs in
+production applications or plugins published to pub.dev.
+
+To try experimental windowing APIs:
+1. Switch to Flutter's main release channel.
+2. Turn on the windowing feature flag.
+
+See: https://github.com/flutter/flutter/issues/30701.
+''';
+
+/// Defines the possible archetypes for a window.
+///
+/// {@template flutter.widgets.windowing.experimental}
+/// Do not use this API in production applications or packages published to
+/// pub.dev. Flutter will make breaking changes to this API, even in patch
+/// versions.
+///
+/// This API throws an [UnsupportedError] error unless Flutter’s windowing
+/// feature is enabled by [isWindowingEnabled].
+///
+/// See: https://github.com/flutter/flutter/issues/30701.
+/// {@endtemplate}
+@internal
+enum WindowArchetype {
+  /// Defines a traditional window
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
```

Comment:
After the template is defined once, you should be able to just use `@macro` now instead of repeating this:

```suggestion
  /// {@macro flutter.widgets.windowing.experimental}
```

Reply:
Ah whoops! Good call :)